openquake/calculators/base.py

```diff
@@ -983,13 +983,20 @@ def import_gmfs(dstore, fname, sids):
     if names[0] == 'rlzi':  # backward compatbility
         names = names[1:]  # discard the field rlzi
     imts = [name[4:] for name in names[2:]]
-    gmf_data_dt = dstore['oqparam'].gmf_data_dt()
-    arr = numpy.zeros(len(array), gmf_data_dt)
-    col = 0
+    oq = dstore['oqparam']
+    if set(oq.imtls) > set(imts):
+        raise ValueError('%s contains %s but the risk model needs %s' %
```

Comment:
Might be more informative to list the missing IMTs in this case.

Reply:
Done

Reply:
Would `set(oq.imtls) - set(imts)` give the missing imts?